### PR TITLE
Create real `NamedTypeSymbol` for private implementation type and use it as a parent for `SynthesizedGlobalMethodSymbol`.

### LINKED ISSUE #70738: Accessing InlineArrays nullable fields with ?? operator exited with code -2146232797
Accessing InlineArrays nullable fields with ?? operator exited with code -2146232797.

**Version Used**: 

.NET 8 RC2 (C# 12)

**Steps to Reproduce**:

```csharp
MyArray array = new();
var element = array[0] ?? 0;

[System.Runtime.CompilerServices.InlineArray(1)]
struct MyArray
{
    private int? _value;
}
```

**Diagnostic Id**:

error MSB6006: "csc.exe" exited with code -2146232797.

**Expected Behavior**:
Compile

**Actual Behavior**:
Build Error

cc. @jaredpar

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #70738.

Since `SynthesizedGlobalMethodSymbol`s are getting used more often, more components run into them while assuming that all good methods have containing type.

### CODE REVIEW

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+        public override int Arity => 0;
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override NamedTypeSymbol ConstructedFrom => this;
+
+        public override bool MightContainExtensionMethods => false;
+
+        public override string Name => _privateImplementationDetails.Name!;
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs:46]: >! [](http://example.com/codeflow?start=73&length=1)

Is `!` needed?

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+        internal override NamedTypeSymbol BaseTypeNoUseSiteDiagnostics => _objectType;
+
+        internal override bool IsRecord => false;
+
+        internal override bool IsRecordStruct => false;
+
+        internal override ObsoleteAttributeData? ObsoleteAttributeData => null;
+
+        public override ImmutableArray<Symbol> GetMembers() => ImmutableArray<Symbol>.Empty;
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs:114]: >ImmutableArray<Symbol>.Empty [](http://example.com/codeflow?start=63&length=28)

Is this correct?

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+        internal override NamedTypeSymbol GetDeclaredBaseType(ConsList<TypeSymbol> basesBeingResolved) => _objectType;
+
+        internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList<TypeSymbol> basesBeingResolved) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers() => ImmutableArray<Symbol>.Empty;
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers(string name) => ImmutableArray<Symbol>.Empty;
+
+        internal override IEnumerable<FieldSymbol> GetFieldsToEmit() => SpecializedCollections.EmptyEnumerable<FieldSymbol>();
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs:149]: >SpecializedCollections.EmptyEnumerable<FieldSymbol>() [](http://example.com/codeflow?start=72&length=53)

Aren't there fields in PrivateImplementationDetails? If there are fields, but this method should not be used, could we `throw Unreachable()`?

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs (hunk 2 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedPrivateImplementationDetailsType.cs:114]: > Is this correct? 

Yes, this is correct. There is no expectation that this API returns synthesized members (there is nothing new in this behavior for symbols) and all members of this type are synthesized.

**Review by cston — APPROVED** [review]

**Comment by RikkiGibson** [comment]
auto-merge might be stuck because integration tests were marked required at the time it was enabled. If it doesn't merge, feel free to disable auto-merge and instead hit merge normally.

**Pull request merged**